### ISSUE TITLE
Axios standardizes headers to be case-insensitive, which causes issues for case-sensitive APIs

### ISSUE THREAD

**Issue by b1z3rr4** [axios/axios#6681]
### Describe the bug

Hi,

I have a question regarding the capitalization of headers in Axios requests. By default, it seems that Axios capitalizes header names (e.g., "content-type" becomes "Content-Type" in some versions).

This behavior can cause issues when working with case-sensitive APIs, such as AWS API Gateway, which require headers to follow specific casing (as noted in the AWS documentation).

I understand that this decision may follow RFC 2616, which states that HTTP header names are case-insensitive. However, this approach can make Axios challenging to use in environments where case-sensitive headers are required.

I’ve noticed other issues opened on this topic, but they have been closed without a final solution, with the decision being to follow the specification:

Issue #4509: https://github.com/axios/axios/issues/4509
Issue #413: https://github.com/axios/axios/issues/413
If there are other related issues, I couldn’t find them here.

Is there any way to configure Axios to send headers in a case-sensitive manner, or is there a workaround to address this issue? Any insights or suggestions would be greatly appreciated!

Thank you!

### To Reproduce

I encountered the error while using AWS pre-signed URLs. The backend of my system generates this upload URL, and the frontend sends the file to the obtained endpoint. My API is written in .NET 8.

### Code snippet

```js
This is an example of a manual PUT request without Axios, where headers are set to be case-sensitive:

PUT /programa_29151_capitulo_9.fdx?AWSAccessKeyId=*****&Expires=1730412840&x-amz-security-token=*****&Signature=***** HTTP/1.1
accept: application/json
content-type: 

In comparison, here’s the same PUT request using Axios:

PUT /programa_11111_capitulo_99.fdx?AWSAccessKeyId=*****&Expires=1730412968&x-amz-security-token=*****&Signature=***** HTTP/1.1
Accept: application/json
Content-Type: 

The only difference between these two requests is the capitalization of the content-type and accept headers, with the manual request having these headers in lowercase (content-type and accept), while Axios automatically capitalizes them (Content-Type and Accept). This is a real example I’ve encountered; the first request succeeds, but the second returns a 403 Forbidden error. I’ve hidden the other request details as they are exactly the same.
```

### Axios Version

1.7.7

### Node.js Version

22.2.0

### Additional Library Versions

```bash
React 18.2.0
```

**Comment by DigitalBrainJS** [comment]
> Is there any way to configure Axios to send headers in a case-sensitive manner, or is there a workaround to address this issue? Any insights or suggestions would be greatly appreciated!

Yeah. The `AxiosHeaders` class used by Axios internally preserves the case of headers. This means that after setting a new header (that was not there before as an object property) Axios won't change its case
```js
import {AxiosHeaders} from 'axios';

const headers = new AxiosHeaders({
  FOO: undefined // just a case preset
});

headers.set({
  'foo': 'fooValue', // it is still FOO
  'bar': 'barValue'
});

console.log(headers.toJSON()); // [Object: null prototype] { FOO: 'fooValue', bar: 'barValue' }
```
Thus, you can create a preset of headers with the desired cases in the instance configuration and Axios will respect it.

```js
const instance = axios.create();

instance.defaults.headers.common = {
  'content-type': undefined,
  accept: 'application/json, text/plain, */*',
  'user-agent': undefined,
  'content-length': undefined,
  AnyCaseYouWant: undefined
}

const {data} = await instance.post(url, 'foo');
```

**Comment by jasonsaayman** [comment]
Updating the docs with this!
